packages/bvaughn-architecture-demo/src/suspense/PointsCache.ts

```diff
@@ -0,0 +1,64 @@
+import { ExecutionPoint, SessionId } from "@replayio/protocol";
+import { client } from "protocol/socket";
+import { unstable_getCacheForType as getCacheForType } from "react";
+
+import { Record, STATUS_PENDING, STATUS_REJECTED, STATUS_RESOLVED, Wakeable } from "../types";
+import { createWakeable } from "../utils/suspense";
+
+// TODO We could add some way for external code (in the client adapter) to pre-populate this cache with known points.
+// For example, when we get Paints they have a corresponding point (and time) which could be pre-loaded here.
+
+type TimeToRecordMap = Map<number, Record<ExecutionPoint>>;
+
+function createMap(): TimeToRecordMap {
+  return new Map();
+}
+
+function getRecordMap(): TimeToRecordMap {
+  return getCacheForType(createMap);
+}
+
+export function getClosestPointForTime(time: number, sessionId: SessionId): ExecutionPoint {
```

Comment:
This is more of a typical Suspense cache implementation. No extra business logic like soft focus, just a map of key (time) to value (execution point) that we build over time– _suspending_ and loading information from the backend when we need to.